pkg/collector/python/datadog_agent.go

```diff
@@ -264,8 +264,11 @@ func lazyInitObfuscator() *obfuscate.Obfuscator {
 		if !cfg.SQLExecPlanNormalize.Enabled {
 			cfg.SQLExecPlanNormalize = defaultSQLPlanNormalizeSettings
 		}
-		if !cfg.Mongo.Enabled {
-			cfg.Mongo = defaultMongoObfuscateSettings
+		if len(cfg.Mongo.KeepValues) == 0 {
+			cfg.Mongo.KeepValues = defaultMongoObfuscateSettings.KeepValues
+		}
+		if len(cfg.Mongo.ObfuscateSQLValues) == 0 {
+			cfg.Mongo.ObfuscateSQLValues = defaultMongoObfuscateSettings.ObfuscateSQLValues
```

Comment:
Before this PR, `cfg` here is not reflected by default values because `DD_APM_OBFUSCATION_*` are not bound to default values. This is the issue that I want to solve with this PR.

After this PR, `cfg.Mongo.Enabled` is `true` since `DD_APM_OBFUSCATION_MONGODB_ENABLED` is bound to `true`. This follows this documentation.

https://github.com/DataDog/datadog-agent/blob/62c67dbe190ebb1128ce7091b1459e2105d896e9/pkg/config/config_template.yaml#L1212-L1215

So, we need this change to keep backward compatibility.

`cfg.SQLExecPlan.Enabled` and `cfg.SQLExecPlanNormalize.Enabled` are `false` by default because `DD_APM_OBFUSCATION_SQL_EXEC_PLAN_ENABLED` and `DD_APM_OBFUSCATION_SQL_EXEC_PLAN_NORMALIZE_ENABLED` are bound to `false`. Therefore, we can keep backward compatibility without code change here.